gen_isr_tables.py does not check that the IRQ number is in bounds

**Describe the bug**
The script creates a fixed size list representing the vector table, and then changes the default values in the list with the appropriate ISR addresses. This works fine when the parameters are correct, but if the code tries to use an IRQ that has a number larger than CONFIG_NUM_IRQS the script crashes.
This happens on this line:
```
vt[irq - offset] = func
```
The list index is not checked before being used.

**To Reproduce**
Steps to reproduce the behavior:
1. Change the default value of CONFIG_NUM_IRQS for any board using it to a value considerably lower than it should be.
2. Alternatively, you can try to add an IRQ with a number larger than the valid configured maximum.
3. Compile

**Expected behavior**
The script should check the index before accessing the list. If the index is out of bounds it should exit with a descriptive error message.

**Impact**
When adding support for new boards, this error can occur if the default value given to CONFIG_NUM_IRQS is wrong, the compilation can crash without outputting a descriptive error message. Also, if a user tries to add an ISR with a large IRQ number, it will crash without a message. This is quite difficult to debug.

**Logs and console output**
```
Traceback (most recent call last):
  File "____/zephyr/arch/common/gen_isr_tables.py",  line 316, in <module>
    main()
  File "____/zephyr/arch/common/gen_isr_tables.py", line 261, in main
    vt[irq - offset] = func
IndexError: list assignment index out of range
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 16.04
 - Toolchain: arm-zephyr-eabi from SDK 0.11.3
 - Commit SHA or Version used: v2.3.0

--- a/arch/common/gen_isr_tables.py
+++ b/arch/common/gen_isr_tables.py
@@ -259,6 +259,9 @@ def main():
             if param != 0:
                 error("Direct irq %d declared, but has non-NULL parameter"
                         % irq)
+            if not 0 <= irq - offset < len(vt):
+                error("IRQ %d (offset=%d) exceeds the maximum of %d" %
+                      (irq - offset, offset, len(vt) - 1))
             vt[irq - offset] = func
         else:
             # Regular interrupt
@@ -302,6 +305,9 @@ def main():
                     debug('IRQ_Pos  = ' + str(irq1))
                     table_index = irq1 - offset
 
+            if not 0 <= table_index < len(swt):
+                error("IRQ %d (offset=%d) exceeds the maximum of %d" %
+                      (table_index, offset, len(swt) - 1))
             if swt[table_index] != (0, spurious_handler):
                 error(f"multiple registrations at table_index {table_index} for irq {irq} (0x{irq:x})"
                       + f"\nExisting handler 0x{swt[table_index][1]:x}, new handler 0x{func:x}"

